haystack/preview/components/file_converters/html.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from typing import List, Optional, Dict, Any, Union
+from pathlib import Path
+
+from haystack.preview.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_to_dict, default_from_dict
+
+with LazyImport("Run 'pip install boilerpy3'") as boilerpy3_import:
+    from boilerpy3 import extractors
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class HTMLToDocument:
+    """
+    A component for converting an HTML file to a Document.
+    """
+
+    def __init__(self, id_hash_keys: Optional[List[str]] = None):
+        """
+        Create a HTMLToDocument component.
+
+        :param id_hash_keys: Generate the Document ID from a custom list of strings that refer to the Document's
+            attributes. Default: `None`
+        """
+        boilerpy3_import.check()
+        self.id_hash_keys = id_hash_keys or []
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(self, id_hash_keys=self.id_hash_keys)
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "HTMLToDocument":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
+
+    @component.output_types(documents=List[Document])
+    def run(self, paths: List[Union[str, Path]], id_hash_keys: Optional[List[str]] = None):
```

Comment:
Sorry I missed this: we don't need id_hash_keys here for now

```suggestion
    def run(self, paths: List[Union[str, Path]]):
```

Reply:
Ok done now with [d7e86be](https://github.com/deepset-ai/haystack/pull/5907/commits/d7e86be7ad4935636f45c2aa924f4641bab6477c) @ZanSara